napalm/base/helpers.py

```diff
@@ -258,52 +258,74 @@ def as_number(as_number_val):
+    match. Regex and other looser matching methods were not implmented to avoid false
+    positive matches. As an example, it would make sense to do "[P|p][O|o]" which would
+    incorrectly match PO = POS and Po = Port-channel, leading to a false positive, not
+    easily troubleshot, found, or known.
+
+    :param interface: The interface you are attempting to expand.
+    :param addl_name_map: A dict containing key/value pairs that is mean to update the base mapping.
+    Used if a OS has specific differences. e.g. {"Po": "PortChannel"} vs {"Po": "Port-Channel"}
+    """
+
+    name_map = {}
+    name_map.update(base_interfaces)
+    interface_type, interface_number = split_interface(interface)
+
+    if isinstance(addl_name_map, dict):
+        name_map.update(addl_name_map)
     # check in dict for mapping
-    if base_interfaces.get(interface_type):
-        long_int = base_interfaces.get(interface_type)
+    if name_map.get(interface_type):
+        long_int = name_map.get(interface_type)
         return long_int + str(interface_number)
-    # if nothing matched, at least return the original
+    # if nothing matched, return the original name
     else:
         return interface
 
 
-def abbreviated_interface_name(interface, update_os_mapping=None):
-    '''
-    Function to retun interface canonical name
-    This puposely does not use regex, or first X characters, to ensure
-    there is no chance for false positives. As an example, Po = PortChannel, and
-    PO = POS. With either character or regex, that would produce a false positive.
-    '''
+def abbreviated_interface_name(interface, addl_name_map=None, addl_reverse_map=None):
+    """Function to return an abbreviated representation of the interface name.
 
-    interface_type, interface_number = int_split_on_match(interface)
+    :param interface: The interface you are attempting to expand.
+    :param addl_name_map (optional): A dict containing key/value pairs that updates
+    the base mapping. Used if an OS has specific differences. e.g. {"Po": "PortChannel"} vs
+    {"Po": "Port-Channel"}
+    :param addl_reverse_map (optional): A dict containing key/value pairs that updates
+    the reverse mapping. Used if an OS has specific differences. e.g. {"PortChannel": "Po"} vs
+    {"PortChannel": "po"}
+    """
 
-    if isinstance(update_os_mapping, dict):
-        base_interfaces.update(update_os_mapping)
-    # check in dict for mapping
-    if base_interfaces.get(interface_type):
-        long_int = base_interfaces.get(interface_type)
-        return reverse_mapping[long_int] + str(interface_number)
-    # if nothing matched, at least return the original
+    name_map = {}
+    name_map.update(base_interfaces)
+    interface_type, interface_number = split_interface(interface)
+
+    if isinstance(addl_name_map, dict):
+        name_map.update(addl_name_map)
+    if isinstance(addl_reverse_map, dict):
+        reverse_mapping.update(addl_reverse_map)
```

Comment:
@itdependsnetworks You can't update `reverse_mapping` as that is modifying the original dictionary and will potentially have unintended behavior if multiple calls are made.

You will need to do something like:

```
name_map_reverse = {}
name_map_reverse.update(reverse_mapping)

# Then...
    if isinstance(addl_reverse_map, dict):
         name_map_reverse.update(addl_reverse_map)

# And update additional references to use name_map_reverse
```